jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -877,4 +891,25 @@ public void dispatchAsync() throws Exception
             servletContextHandler.requestDestroyed(servletContextRequest, servletApiRequest);
         }
     }
+
+    public void readComplete()
+    {
+        if (_completeCount.decrementAndGet() == 0 && _completeListener != null)
+        {
+            _completeListener.complete();
+        }
+    }
+
+    public void writeComplete()
+    {
+        if (_completeCount.decrementAndGet() == 0 && _completeListener != null)
+        {
+            _completeListener.complete();
+        }
+    }
+
+    public void setCompleteListener(CompleteListener listener)
+    {
+        _completeListener = listener;
+    }
```

Comment:
This whole listener mechanism is not needed.   See my comment in ServletApiRequest